File: ClaOJ_Charity_Marathon_2023/H.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
#define pb push_back
#define pf push_front
#define db double
#define X first
#define Y second
#define fast ios_base::sync_with_stdio(false);cout.tie(NULL);cin.tie(NULL);
#define all(x) (x).begin(),(x).end()
#define rall(x) (x).rbegin(),(x).rend()
#define nl '\n'
#define ii pair<int,int>

const int M=1e9+7;
const int N = 1e6+5;

int moveX[] = {0, 0, 1, -1};
int moveY[] = {1, -1, 0, 0};

void indef(){
	#ifndef ONLINE_JUDGE
		freopen("input.txt","r",stdin);
		freopen("output.txt","w",stdout);
	#endif
}
int a[N];
int mp[N];
void solve(){
	int n;
	cin >> n;
	int mx = 0;
	int ans = 0;
	for(int i=1;i<=n;i++) cin >> a[i];
	for(int i=1;i<=n;i++){
		cout << mp[a[i]] << ' ' << i << ' ';
		if(mp[a[i]] && mx!=i) ans += i - mx + 1;
		if(mp[a[i+1]]) mx = max(mx,mp[a[i+1]]);
		cout << ans << nl;
		mp[a[i]] = i;
	}

	
	cout << ans;
}
int main(){
	fast;
	indef();
	int tt=1;
	//cin >> tt;
	while(tt--) solve();
}

File: ICPC/2023/ICPCSouth2022/H.py

import sys
from os import path
if(path.exists('input.txt')):
    sys.stdin = open("input.txt","r")
    sys.stdout = open("output.txt","w")

def multi(n):
    ans=1
    while(n>0):
        ans*=(n%10)
        n//=10
        if(ans==0): return ans
    return ans
l=input()
r=input()

a=b=ans=0
M=int(1e9+7)
if(len(l)<len(r)):
    l=""
    tmp=r[0]
    tmp=ord(tmp)-1
    l+=chr(tmp) 
    for i in range(1,len(r)):
        l+='9'
else: 
    if(ord(l[0])<ord(r[0])):
        l=""
        tmp=r[0]
        tmp=ord(tmp)-1
        l+=chr(tmp)
        for i in range(1,len(r)):
            l+='9'
a=int(l)
b=int(r)
print(l,r)
for i in range(a,b+1):
    ans=max(ans,multi(i))
print(ans%M)
